tests/php/src/Validation/ScannableURLProviderTest.php

```diff
@@ -0,0 +1,392 @@
+						[
+							'post_type' => $post_type,
+						]
+					)
+				);
+			}
+
+			$actual_posts = $this->call_private_method( $this->validation_url_provider, 'get_posts_by_type', [ $post_type ] );
+			$this->assertEquals( $expected_posts, array_values( $actual_posts ) );
+
+			// Test with the $offset and $number arguments.
+			$offset       = 0;
+			$actual_posts = $this->call_private_method( $this->validation_url_provider, 'get_posts_by_type', [ $post_type, $offset, $number_posts_each_post_type ] );
+			$this->assertEquals( array_slice( $expected_posts, $offset, $number_posts_each_post_type ), $actual_posts );
+		}
+	}
+
+	/**
+	 * Test get_taxonomy_links.
+	 *
+	 * @covers ::get_taxonomy_links()
+	 */
+	public function test_get_taxonomy_links() {
+		$number_links_each_taxonomy = 20;
+		$taxonomies                 = get_taxonomies(
+			[
+				'public' => true,
+			]
+		);
+
+		foreach ( $taxonomies as $taxonomy ) {
+			// Begin the expected links with the term links that already exist.
+			$expected_links             = array_map( 'get_term_link', get_terms( [ 'taxonomy' => $taxonomy ] ) );
+			$terms_for_current_taxonomy = [];
+			for ( $i = 0; $i < $number_links_each_taxonomy; $i++ ) {
+				$terms_for_current_taxonomy[] = self::factory()->term->create(
+					[
+						'taxonomy' => $taxonomy,
+					]
+				);
+			}
+
+			// Terms need to be associated with a post in order to be returned in get_terms().
+			$result = wp_set_post_terms(
+				self::factory()->post->create(),
+				$terms_for_current_taxonomy,
+				$taxonomy
+			);
+			$this->assertFalse( is_wp_error( $result ) );
+
+			$expected_links  = array_merge(
+				$expected_links,
+				array_map( 'get_term_link', $terms_for_current_taxonomy )
+			);
+			$number_of_links = 100;
+			$actual_links    = $this->call_private_method( $this->validation_url_provider, 'get_taxonomy_links', [ $taxonomy, 0, $number_of_links ] );
+
+			// The get_terms() call in get_taxonomy_links() returns an array with a first index of 1, so correct for that with array_values().
+			$this->assertEquals( $expected_links, array_values( $actual_links ) );
+			$this->assertLessThan( $number_of_links, count( $actual_links ) );
+
+			$number_of_links           = 5;
+			$offset                    = 10;
+			$actual_links_using_offset = $this->call_private_method( $this->validation_url_provider, 'get_taxonomy_links', [ $taxonomy, $offset, $number_of_links ] );
+			$this->assertEquals( array_slice( $expected_links, $offset, $number_of_links ), array_values( $actual_links_using_offset ) );
+			$this->assertEquals( $number_of_links, count( $actual_links_using_offset ) );
+		}
+	}
+
+	/**
+	 * Test get_search_page.
+	 *
+	 * @covers ::get_search_page()
+	 */
+	public function test_get_search_page() {
+		// Normally, this should return a string, unless the user has opted out of the search template.
+		$this->assertTrue( is_string( $this->call_private_method( $this->validation_url_provider, 'get_search_page' ) ) );
```

Comment:
The `assertIsString()` assertion seems more appropriate.

```suggestion
		$this->assertIsString( $this->call_private_method( $this->validation_url_provider, 'get_search_page' ) );
```